packages/contractkit/src/identity/offchain-data-wrapper.test.ts

```diff
@@ -1,63 +1,102 @@
 
-  const writer = ACCOUNT_ADDRESSES[0]
-  const signer = ACCOUNT_ADDRESSES[1]
+  const writerPrivate = ACCOUNT_PRIVATE_KEYS[0]
+  const writerPublic = privateKeyToPublicKey(writerPrivate)
+  const writerAddress = publicKeyToAddress(writerPublic)
+
+  const readerPrivate = ACCOUNT_PRIVATE_KEYS[1]
+  const readerPublic = privateKeyToPublicKey(readerPrivate)
+  const readerAddress = publicKeyToAddress(readerPublic)
+
+  const reader2Private = ACCOUNT_PRIVATE_KEYS[2]
+  const reader2Public = privateKeyToPublicKey(reader2Private)
+  const reader2Address = publicKeyToAddress(reader2Public)
+
+  const signerPrivate = ACCOUNT_PRIVATE_KEYS[3]
+  const signerPublic = privateKeyToPublicKey(signerPrivate)
+  const signerAddress = publicKeyToAddress(signerPublic)
+
+  const writerEncryptionKeyPrivate = ensureLeading0x(randomBytes(32).toString('hex'))
+  const writerEncryptionKeyPublic = privateKeyToPublicKey(writerEncryptionKeyPrivate)
+  const readerEncryptionKeyPrivate = ensureLeading0x(randomBytes(32).toString('hex'))
+  const readerEncryptionKeyPublic = privateKeyToPublicKey(readerEncryptionKeyPrivate)
+
   const WRITER_METADATA_URL = 'http://example.com/writer'
   const WRITER_STORAGE_ROOT = 'http://example.com/root'
-  const WRITER_LOCAL_STORAGE_ROOT = `/tmp/offchain/${writer}`
+  const WRITER_LOCAL_STORAGE_ROOT = `/tmp/offchain/${writerAddress}`
   let accounts: AccountsWrapper
   let wrapper: OffchainDataWrapper
+  let readerWrapper: OffchainDataWrapper
+  let reader2Wrapper: OffchainDataWrapper
 
   beforeEach(async () => {
     accounts = await kit.contracts.getAccounts()
+    await accounts.createAccount().sendAndWaitForReceipt({ from: writerAddress })
 
-    await accounts.createAccount().sendAndWaitForReceipt({ from: writer })
+    await accounts
+      .setAccountDataEncryptionKey(writerEncryptionKeyPublic)
+      .sendAndWaitForReceipt({ from: writerAddress })
 
-    const metadata = IdentityMetadataWrapper.fromEmpty(writer)
+    const metadata = IdentityMetadataWrapper.fromEmpty(writerAddress)
     await metadata.addClaim(
       createStorageClaim(WRITER_STORAGE_ROOT),
-      NativeSigner(web3.eth.sign, writer)
+      NativeSigner(web3.eth.sign, writerAddress)
     )
 
     fetchMock.mock(WRITER_METADATA_URL, metadata.toString())
-    await accounts.setMetadataURL(WRITER_METADATA_URL).sendAndWaitForReceipt({ from: writer })
+    await accounts
+      .setMetadataURL(WRITER_METADATA_URL)
+      .sendAndWaitForReceipt({ from: writerAddress })
+
+    kit.addAccount(writerPrivate)
+
+    wrapper = new OffchainDataWrapper(writerAddress, kit)
+    wrapper.storageWriter = new MockStorageWriter(
+      WRITER_LOCAL_STORAGE_ROOT,
+      WRITER_STORAGE_ROOT,
+      fetchMock
+    )
+
+    readerWrapper = new OffchainDataWrapper(readerAddress, kit)
+    reader2Wrapper = new OffchainDataWrapper(reader2Address, kit)
   })
 
-  afterEach(async () => {
+  afterEach(() => {
     fetchMock.reset()
+    kit.removeAccount(writerAddress)
```

Comment:
Related to my comment on `kit.ts`, instead of using `removeAccount` here, it make be best to simply make a new `kit` in the `beforeEach`.

Reply:
IIRC, the problem is that the account persists on a new kit instance since its on `web3`?